Omega stronghold rift now go to Hershfield main terminal

clouths opened the pull request:
Closes: [OpenPerpetuum/OP-Project#139](https://github.com/OpenPerpetuum/OP-Project/issues/139)

I used hardcoded value for the teleport location value to be at docking range for the Hershfield main terminal.

clouths committed:
Change the exit rift of Omega Stronghold to go to Hershrfield main
terminal

MikeJeffers commented on src/Perpetuum/Services/RiftSystem/Rift.cs:
```diff
@@ -194,13 +194,12 @@ public void UseItem(Player player)
+                var destZone = player.Character.GetZone(8); // Hershfield zone
                 var teleport = _teleportStrategyFactories.TeleportToAnotherZoneFactory(destZone);
-                teleport.TargetPosition = teleportColumns.First().CurrentPosition;
+                teleport.TargetPosition = new Position(964, 804); // At the Hershfield main terminal
```
Try using a RandomWalkablePosition around this point - it will more safely resolve a place a player may be able to walk out of, in case things about the zone/layers are changed.

MikeJeffers replied:
Actually scratch that: following loyals comments - find the DockingBase itself off of the zone directly, and call methods on it to resolve a position
Like this: https://github.com/OpenPerpetuum/PerpetuumServer/blob/Development/src/Perpetuum/Units/DockingBases/DockingBase.cs#L30

There are ways to lookup "units" on a zone by type - much like the teleport columns.  You will need to do that for DockingBases, and look for the first/only one that is available on zone 8

clouths replied:
I added a new commit but using DockingBaseHelper is adding a new dependency to the Rift class.

clouths committed:
* Change the hardcoded position for the hershfield main terminal
destination to the same logic as undocking from the station in terms of
the target location of the teleport

MikeJeffers commented on src/Perpetuum/Services/RiftSystem/Rift.cs:
```diff
@@ -199,7 +202,8 @@ public void UseItem(Player player)
                 var destZone = player.Character.GetZone(8); // Hershfield zone
                 var teleport = _teleportStrategyFactories.TeleportToAnotherZoneFactory(destZone);
-                teleport.TargetPosition = new Position(964, 804); // At the Hershfield main terminal
+                var dockingBase = _dockingBaseHelper.GetDockingBase(142); // Hershfield main terminal
```
Ok while this does get the terminal as intended - we now have 2 hardcoded values and a new dependency.
Seeing this now, let's see if we can avoid one of these and "find" the other.
The zone has the terminal as a Unit on it...
Or
The dockingbase is surely a member of a Zone...
If you can resolve the Zone from the DockingBase - that would probably be the best way forward.

Whichever strategy you pick - as long as we dont have to know the ID of both the zone AND the terminal - that would be best.

clouths replied:
I updated the code, but the only thing I am not sure in terms of performance is the DockingBase explicit cast.

clouths replied:
It turns out, the OfType method was more convenient. I did an other commit.

MikeJeffers commented:
Really good stuff!  One last tweak on your approach and this is a really solid fix for our messy old code!

clouths committed:
* Replace the usage of DockingBaseHelper with a search of the Units
within the zone to get the DockingBase

clouths committed:
Use OfType method to get the docking base, since it is more convenient

MikeJeffers reviewed (approved):
Awesome - Very clean, maintainable and works as intended!
